services/core-api/app/api/mines/mine/models/mine.py

```diff
@@ -134,29 +134,36 @@ def find_by_mine_no(cls, _id):
         return cls.query.filter_by(mine_no=_id).filter_by(deleted_ind=False).first()
 
     @classmethod
-    def find_by_mine_name(cls, term=None):
+    def find_by_mine_name(cls, term=None, major=None):
         MINE_LIST_RESULT_LIMIT = 50
         if term:
             name_filter = Mine.mine_name.ilike('%{}%'.format(term))
             mines_q = Mine.query.filter(name_filter).filter_by(deleted_ind=False)
-            mines = mines_q.limit(MINE_LIST_RESULT_LIMIT).all()
         else:
-            mines = Mine.query.limit(MINE_LIST_RESULT_LIMIT).all()
-        return mines
+            mines_q = Mine.query
+
+        if major is not None:
+            mines_q = mines_q.filter_by(major_mine_ind=major)
+            current_app.logger.debug(f'FILTER BY MAJOR = {major}')
+        return mines_q.limit(MINE_LIST_RESULT_LIMIT).all()
 
     @classmethod
-    def find_by_name_no_permit(cls, term=None):
+    def find_by_name_no_permit(cls, term=None, major=None):
         MINE_LIST_RESULT_LIMIT = 50
         if term:
             name_filter = Mine.mine_name.ilike('%{}%'.format(term))
             number_filter = Mine.mine_no.ilike('%{}%'.format(term))
             permit_filter = Permit.permit_no.ilike('%{}%'.format(term))
             mines_q = Mine.query.filter(name_filter | number_filter).filter_by(deleted_ind=False)
             permit_q = Mine.query.join(Permit).filter(permit_filter)
-            mines = mines_q.union(permit_q).limit(MINE_LIST_RESULT_LIMIT).all()
+            mines_q = mines_q.union(permit_q)
```

Comment:
If you put the `permit_q` declaration in front of the initial `mines_q` declaration you can assign this in one line.